test-executor/src/main/java/com/cognifide/aet/executor/SuiteServlet.java

```diff
@@ -104,6 +113,20 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
     }
   }
 
+  private String getChecksumFromJson(Map<String, String> requestData) {
+    String checkSum = StringUtils.EMPTY;
+    JsonParser parser = new JsonParser();
+    String projectChecksSum = requestData.get(PROJECT_CHECK_SUM);
+    JsonElement parse = parser.parse(projectChecksSum);
+    if (parse.isJsonObject()) {
+      JsonObject asJsonObject = parse.getAsJsonObject();
+      if (asJsonObject.isJsonObject()) {
+        checkSum = asJsonObject.get(PROJECT_CHECK_SUM).getAsString();
```

Comment:
The user will get NullPointerException here.
You are trying to extract checksum from `asJsonObject`, providing the key `"projectChecksSum"`, but the checksum property in the json which is returned by checksum servlet (/bin/checksum/combined)
you created in https://bitbucket.cognifide.com/projects/RND/repos/aet-checksum-generator/browse
is named `projectCheckSum` (check vs check**s**). So the `.get()` returns null and following `.getAsString()` throws NPE.

Due to this, the code in present form can never actually work.
The name of the property needs to be unified. Changed either in here or in `AetChecksumGenerator` project.